### PR TITLE
Remove the limit of agents that a manager can support

### PR DESCRIPTION
**Pull request by TomasTurina**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/6097|

## Description

This PR makes Wazuh independent of the number configured for the `MAX_AGENTS` limit.

This limit has been deprecated in the following components:

- **Remoted**: We have improved the RIDS logic so that now the number of open file handlers does not depend on the agent limit and depends on the number of agents connected at any given time. In addition, we have replaced the pending message queue, which was initialized with the size of the agent limit, with a linked queue that increases with the number of connected agents.

- **Rootcheck**: We have improved the mechanism that rootcheck uses to store information from agent scans. Now, each log is stored directly in the `pm_event` table of the corresponding agent through Wazuh DB. The files located at `/queue/rootcheck`, which were being tracked with arrays initialized with the agent size limit, are now not needed so they have been removed.

- **Upgrade module**: The upgrade queue had an initial size of the agent limit. Now, this queue has been replaced with a linked queue that increases with the number of agents queued to upgrade.

## Configuration options

```xml
<remote>
   <rids_closing_time>10m</rids_closing_time>
</remote>
```

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Source upgrade
- [x] Review logs syntax and correct language
- [x] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)

- [x] Added unit tests (for new features)

### CODE REVIEW

**Review by vikman90 — CHANGES_REQUESTED** [review]

File: etc/internal_options.conf [hunk 1]
```diff
@@ -342,7 +342,7 @@ wazuh_database.sync_agents=1
 wazuh_database.sync_syscheck=0
 
 # Synchronize policy monitoring data with Rootcheck database
-wazuh_database.sync_rootcheck=1
+wazuh_database.sync_rootcheck=0
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply etc/internal_options.conf]: Removed in https://github.com/wazuh/wazuh/pull/6450/commits/44dd26da96548416e2f0e1ef0e2c57f8d1ac26f7

File: src/wazuh_modules/wm_database.c [hunk 2]
```diff
@@ -961,7 +867,7 @@ wmodule* wm_database_read() {
 
     data.sync_agents = getDefine_Int("wazuh_database", "sync_agents", 0, 1);
     data.sync_syscheck = 0; //getDefine_Int("wazuh_database", "sync_syscheck", 0, 1);
-    data.sync_rootcheck = getDefine_Int("wazuh_database", "sync_rootcheck", 0, 1);
+    data.sync_rootcheck = 0; //getDefine_Int("wazuh_database", "sync_rootcheck", 0, 1);
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply src/wazuh_modules/wm_database.c]: Removed in https://github.com/wazuh/wazuh/pull/6450/commits/44dd26da96548416e2f0e1ef0e2c57f8d1ac26f7

File: src/config/remote-config.c [hunk 3]
```diff
@@ -23,6 +23,7 @@ int Read_Remote(XML_NODE node, void *d1, __attribute__((unused)) void *d2)
     unsigned int deny_size = 1;
     remoted *logr;
     int defined_queue_size = 0;
+    const int default_rids_closing_time = 300;
```
Comment (vikman90) [inline src/config/remote-config.c:26]: This is a constant definition, it should be in capitals.

File: src/config/authd-config.c [hunk 4]
```diff
@@ -129,18 +128,11 @@ int Read_Authd(XML_NODE node, void *d1, __attribute__((unused)) void *d2) {
...
         } else if (!strcmp(node[i]->element, xml_limit_maxagents)) {
-            short b = eval_bool(node[i]->content);
-
-            if (b < 0) {
-                merror(XML_VALUEERR, node[i]->element, node[i]->content);
-                return OS_INVALID;
-            }
-
-            config->flags.register_limit = b;
+            mwarn("The <%s> tag is deprecated since version 4.1.0.", xml_limit_maxagents);
```
Comment (vikman90) [inline src/config/authd-config.c:131]: This line causes a warning log after upgrading.

**Review by vikman90 — CHANGES_REQUESTED** [review]

**Review by vikman90 — APPROVED** [review]

**Pull request merged**